### PR TITLE
BlockLocator: deprecate add & no-args constructor

### PR DESCRIPTION
**Pull request by msgilligan**
Deprecate .add() and no-args constructor in favor of providing complete list of hashes at creation time.

Update all usages to use the alternative methods.

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/Peer.java [hunk 1]
```diff
@@ -1437,19 +1436,21 @@ private void blockChainDownloadLocked(Sha256Hash toHash) {
...
+            hashes.add(cursor.getHeader().getHash());
             try {
                 cursor = cursor.getPrev(store);
             } catch (BlockStoreException e) {
                 log.error("Failed to walk the block chain whilst constructing a locator");
                 throw new RuntimeException(e);
             }
         }
-        // Only add the locator if we didn't already do so. If the chain is < 50 blocks we already reached it.
+        // Only add genesis hash to locator if we didn't already do so. If the chain is < 100 blocks we already reached it.
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/Peer.java:1450]: Is this comment change related to the change of BlockLocator construction? It seems to me it's unrelated.

**Review by schildbach — COMMENTED** [review]
Looks good, however one question inline.

File: core/src/main/java/org/bitcoinj/core/Peer.java (hunk 1 above)
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/Peer.java:1450]: Unrelated. I just noticed it was wrong and fixed it. Do you want that in a separate commit?

Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/Peer.java:1450]: See PR #3151

**Comment by msgilligan** [comment]
@schilbach rebased after PR #3151 was merged.

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**